react-renderer/src/inspector/views/Mod.tsx

```diff
@@ -0,0 +1,161 @@
+import * as React from "react";
+import IViewProps from "./IViewProps";
+import AttrPicker from "./mod/AttrPicker";
+import { staticMap } from "src/componentMap";
+import styled from "styled-components";
+import defmap from "./mod/defmap";
+
+const ShapeItem = styled.li<any>`
+  display: block;
+  padding: 1em;
+  margin-top: -1px;
+  border: 1px solid #d1d1d1;
+  background-color: ${({ selected }: any) =>
+    selected ? "#F9F9F9" : "#f0f0f0"};
+  color: rgba(0, 0, 0, 0.5);
+  font-family: monospace;
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  cursor: pointer;
+`;
+
+interface IState {
+  selectedShape: number;
+}
+
+class Mod extends React.Component<IViewProps, IState> {
+  public readonly state = { selectedShape: -1};
+  public setSelectedShape = (key: number) => {
+    this.setState({ selectedShape: key});
+  };
+  public modAttr = (attrname: string, attrval: Value<any>) => { 
+    const {frame, modShapes, frameIndex, history} = this.props;
+    const {selectedShape} = this.state;
+    if (frameIndex === -1 || frameIndex === history.length - 1) { // make it only work on last frame
+      const shap = frame!.shapes[(selectedShape === -1 ? 0 : selectedShape)]; // the -1 index behaves weirdly so workaround
```

Comment:
would be interested in why:

* `-1` index behaves weirdly

* `frame!` is possibly null: just check if it's null at the top level of the function?